src/subcommand/init.rs

```diff
@@ -0,0 +1,104 @@
+    result::WrapOk,
+};
+use error_stack::Result;
+use indoc::formatdoc;
+
+/// Errors encountered during init.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    /// A config file already exists
+    #[error("config file exists")]
+    ConfigFileExists,
+
+    /// Creating the data root directory
+    #[error("create data root")]
+    CreateDataRoot(PathBuf),
+
+    /// Writing the file did not work
+    #[error("write config file")]
+    WriteConfigFile(PathBuf),
+}
+
+/// generate the config and db files in the default location
+#[tracing::instrument(skip_all)]
+pub fn main(data_root: &PathBuf) -> Result<(), Error> {
+    let default_already_exists = write_config(data_root, "config.yml", false)?;
+    write_config(data_root, "config.example.yml", true)?;
+    if default_already_exists {
+        let output = formatdoc! {r#"
+
+        `broker init` detected a previously existing config file at {config} and left it as is.
+
+        `broker init` did, however, create a new example config file for you at {example_config}.
+
+        This example config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    } else {
+        let output = formatdoc! {r#"
+
+        `broker init` created an example config in {config}.
+
+        The config file contains a detailed explanation of everything you need to do to get broker up and running.
+
+        The next step is to open {config} and follow the instructions to configure broker.
+
+        We also wrote the same example file to {example_config} to serve as a reference for you in the future.
+
+        You can safely re-run `broker init` at any time to re-generate the "config.example.yml" file.
+
+        "#,
+            config = data_root.join("config.yml").display(),
+            example_config = data_root.join("config.example.yml").display(),
+        };
+        println!("{}", output);
+    }
+    Ok(())
+}
+
+fn write_config(data_root: &PathBuf, filename: &str, force_write: bool) -> Result<bool, Error> {
+    let config_file_path = data_root.join(filename);
+    if config_file_path.try_exists().unwrap_or(false) && !force_write {
+        return true.wrap_ok();
+    }
+
+    std::fs::create_dir_all(data_root)
+        .context_lazy(|| Error::CreateDataRoot(data_root.clone()))
+        .help_lazy(|| {
+            formatdoc! {r#"
+        `broker init` encountered an error while attempting to create the config directory {}.
+        This can happen if you do not have permission to create the directory.
+        Please ensure that you can create a directory at this location and try again
+        "#, data_root.display()}
+        })?;
```

Comment:
They already know it's an error; prefer to not add noise for the user. `help` and associated methods are meant to give users actionable advice to resolve this issue themselves. To add context on why Broker is doing a thing, use `describe`; to add context on what specifically it was doing when the error occurred, attach the information to the actual error.

Also please indent this to match the block, and since this is a multiline block add punctuation so it reads as a sentence. Finally, you don't have any characters in the string which require making this a raw string.

Also minor nit: instead of `clone`, prefer to use a method that encodes the semantics you're after. Here I'd use `to_owned` or `to_path_buf`, because they describe that I'm looking to create a `PathBuf` from the provided `&Path`, not clone the pointer.

Example:
```
/// Errors encountered during init.
#[derive(Debug, thiserror::Error)]
pub enum Error {
    /// Creating the data root directory
    #[error("create data root at {}", .0.display())]
    CreateDataRoot(PathBuf),
}

// ... in your function ...

std::fs::create_dir_all(data_root)
    .context_lazy(|| Error::CreateDataRoot(data_root.to_owned()))
    .describe_lazy(|| formatdoc! {"
        Broker requires that the data root exists and is a directory in order to create example config files.
        If it does not exist, Broker attempts to create it at a default location for your user.
    "})
    .help_lazy(|| formatdoc! {"
        This can happen if Broker did not have permission to create the directory.
        Try creating the directory yourself then running Broker again.
        Alternately, you may specify a different data root: run Broker with the `-h` argument to see how.
    "})?;
```